pkg/gardenadm/resources.go

```diff
@@ -90,23 +94,33 @@ func ReadManifests(log logr.Logger, fsys fs.FS) (*gardencorev1beta1.CloudProfile
 					return fmt.Errorf("found more than one *gardencorev1beta1.Shoot resource, but only one is allowed")
 				}
 				shoot = typedObj
+
+			case *gardencorev1beta1.ControllerRegistration:
+				controllerRegistrations = append(controllerRegistrations, typedObj)
+
+			case *gardencorev1.ControllerDeployment:
+				controllerDeployments = append(controllerDeployments, typedObj)
 			}
 		}
 
 		return nil
 	}); err != nil {
-		return nil, nil, nil, fmt.Errorf("failed reading Kubernetes resources from config directory: %w", err)
+		err = fmt.Errorf("failed reading Kubernetes resources from config directory: %w", err)
+		return
```

Comment:
Might be personal preference, but I am not a big fan of naked `return`s. Therefore, I prefer the following

```suggestion
		return nil, nil, nil, nil, nil, fmt.Errorf("failed reading Kubernetes resources from config directory: %w", err)
```

(same for the lines below)